src/apps/backend/modules/documentation/documentation-service.ts

```diff
@@ -0,0 +1,271 @@
+import fs from 'fs';
+import path from 'path';
+
+import { ConfigService } from '../config';
+import { HttpRoute } from '../list-routes';
+import { OpenAIService } from '../openai';
+
+import { DOCUMENTATION_GENERATION_PROMPT } from './internals/constants';
+import {
+  ErrorReadingFile,
+  HttpRouteWithControllerAndSerializerDetails,
+  HttpRouteWithRootFolderPath,
+  MarkdownDocumentation,
+} from './types';
+
+export default class DocumentationService {
+  public static expressRoutesList: HttpRouteWithRootFolderPath[] = [];
+  public static restApiFolderPath = 'rest-api';
+  public static controllerFileSuffix = '-controller.js';
+  public static routerFileSuffix = '-router.js';
+  public static serializerFileSuffix = '-serializer.js';
+
+  public static async getDocumentation(): Promise<MarkdownDocumentation> {
+    const isDocumentationEnabled = ConfigService.getValue<boolean>(
+      'documentation.enabled',
+    );
+
+    if (!isDocumentationEnabled) {
+      return {
+        markdownDocumentation: '',
+      };
+    }
+
+    const routes = this.getRoutesWithControllerAndSerializerDetails();
+    const prompt = `${DOCUMENTATION_GENERATION_PROMPT}\n${JSON.stringify(routes, null, 2)}`;
+    return {
+      markdownDocumentation:
+        await OpenAIService.getChatCompletionResponse(prompt),
+    };
+  }
+
+  private static getRoutesWithControllerAndSerializerDetails(): HttpRouteWithControllerAndSerializerDetails[] {
```

Comment:
In an ideal world, it would be good to move these into a new file - `DocumentGenerator.ts` inside internals and have the service call into it.

This is causing the service file to bloat and exposes a lot more internal implementation details when someone looks at the file.

Reply:
Moved this and respective private methods to `document-generator-util.ts` file sir.